### DISCUSSION TITLE
Sign Message not work in SafePal App

### DISCUSSION THREAD

**Discussion by AdemKao** [WalletConnect/walletconnect-monorepo#3103] [Developer Support]
Hello,
I have been testing the sign message feature between WalletConnectV2 and SafePal App using the following websites:
[lab.web3modal.com](https://lab.web3modal.com/with-wagmi/react)
[wagmi](https://wagmi.sh/examples/sign-message)

However, I encountered an issue where there is no pop-up screen appearing on SafePal. 
The device version  which I used:
device: iPhone SE
os: 16.5
SafePal: 4.0.2

I'm ensure  whether  this issue is related on WalletConnectV2 or if SafePal does not currently support the WalletConnectV2 signing feature. I would greatly appreciate any assistance or insights regarding this matter.  

Thanks in advance for your reply.

**Comment by boidushya** [comment]
@AdemKao we tested this on react-app.walletconnect.com and were unable to reproduce this with `personal_sign`

Have you made sure 
- You're on the latest version of SafePal
- You're not using `eth_sign` RPC method since its a [known](https://support.metamask.io/hc/en-us/articles/14764161421467-What-is-eth-sign-and-why-is-it-a-risk-) security issue

<img width="997" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/26746725/a777a9fb-b9be-4dce-a645-e876f2e0c7e3">

**Reply by TEE2DWHY** [reply]
@boidushya. Thanks for your comment. I do have an issue. I noticed the method sendTransaction does not work in trust wallet but it works in safepal and metamask. I did wrap my provider inside ethers. Any solution?🙏🏼

**Reply by boidushya** [reply]
Ah if this is wallet specific, I'm afraid this might be an issue with Trust Wallet's implementation rather than WalletConnect. It might be worthwhile to know that if you're testing using testnets, Trust Wallet does not support them.

**Reply by TEE2DWHY** [reply]
I’m actually testing on a mainnet. And it’s not been positive

**Reply by boidushya** [reply]
What type of error are you getting on trust wallet for sendTransaction?

**Reply by TEE2DWHY** [reply]
@boidushya. This is the error: node_modules/@walletconnect/relay-api/src/jsonrpc.ts not found #2851

**Reply by boidushya** [reply]
Are you by any chance not on the latest version? And are you facing this issue on react-app.walletconnect.com?

**Reply by TEE2DWHY** [reply]
@boidushya Thank you. I just tested the website link you sent “react-app.walletconnect.com” and try send transaction functionality worked . Can I get access to the code sample?

**Reply by boidushya** [reply]
If you're using wagmi, this should help you out!
https://wagmi.sh/examples/send-transaction

**Reply by TEE2DWHY** [reply]
I tried the send transaction function on WAGMI directly on trust wallet but it’s the same thing. How ever, when I tested this: “react-app.walletconnect.com” it worked fine. I’m presently trying to look for a similar code base with the send transaction function. Could you help with a repo @boidushya
